dbt_platform_helper/domain/versioning.py

```diff
@@ -65,3 +78,85 @@ def check_platform_helper_version_mismatch(self):
                 f"v{required_version} specified for the project."
             )
             self.io.warn(message)
+
+    def check_if_needs_update(self):
+        if self.skip_versioning_checks:
+            return
+
+        version_status = self._get_version_status(include_project_versions=False)
+
+        message = (
+            f"You are running platform-helper v{version_status.local}, upgrade to "
+            f"v{version_status.latest} by running run `pip install "
+            "--upgrade dbt-platform-helper`."
+        )
+
+        try:
+            version_status.local.validate_compatibility_with(version_status.latest)
+        except IncompatibleMajorVersionException:
+            self.io.error(message)
+        except IncompatibleMinorVersionException:
+            self.io.warn(message)
+
+    def _get_version_status(
+        self,
+        include_project_versions: bool = True,
+    ) -> PlatformHelperVersionStatus:
+        try:
+            locally_installed_version = self.local_version_provider.get_installed_tool_version(
+                "dbt-platform-helper"
+            )
+        except LocalVersionProviderException:
```

Comment:
Codecov raised that l109-110 wasn't tested. However, I'm not sure if this would ever be executed? Presumably if the locally installed version can't be found then the execution shouldn't progress further? Or if it should continue, then some of the warning messages we show to the user needs to be tweaked as we're directly referencing `version_status.local` (e.g. instead of showing "You are running platform-helper {local_version} against {required_version", we display another error message)

Reply:
Discussed with Kate, removed this except block as it's unnecessary